### ISSUE TITLE
RustDesk Touch Input/Service Enable/Disable with Tasker

### ISSUE THREAD

**Issue by homeusepb** [rustdesk/rustdesk#3322]
### Description

So I use RustDesk for automating my grandparents and parent's phone. I want to setup unattended access and obviously as you guys know, you do not allow RustDesk to start on boot (why it is beyond me, its a simple feature that is needed - but I tried modifying your permissions and I believe if you enable RECEIVE BOOT COMPLETED, it crashes because it needs to call some other methods properly which you probably have not yet found out how to?).

Anyways, so I was thinking why not make an intent or some form of automation that something like Tasker or Macrodroid can send a command to trigger? I can setup a profile or webhook for these automation apps and in turn these apps like tasker can send RustDesk an intent or some form of launch a service to automatically trigger Input/Main Service in background?

Currently only way is to touch and interact with screen, but that does not work for locked phones, despite it is very unreliable.

### Impact

The main benefit is that AnyDesk and TeamViewer and even your Droid-NG (what you guys used to get started from on Android) supports auto boot and mainly intents to launch services with.

The inability to have unattended access that works reliably is holding many business use cases back (although Im not a business, but many businesses I have seen discuss this).

**Comment by rustdesk** [comment]
@Heap-Hop

**Comment by rustdesk** [comment]
![image](https://user-images.githubusercontent.com/71636191/220516470-7f16dc17-3152-4b86-91e1-c8e1f90e36f9.png)

**Comment by rustdesk** [comment]
If need, we should change the way we capture screen now.

**Comment by homeusepb** [comment]
I have seen all these issues, and I know many duplicates have been made, but on the only-open topic (https://github.com/rustdesk/rustdesk/discussions/1516), nobody has really tried to fix the crashing issue if RECEIVE BOOT permission is enabled, and nobody has tried to use a webhook or Android intents to control service start. So this topic is not really a duplicate.

**Comment by homeusepb** [comment]
Currently I've tested RustDesk on Android from 1.1.9 to 1.20 nightly, it works well but two issues;
1. No boot service start OR ANY remote method to turn it on
2. It sometimes fails to connect to self hosted/public server because the UDP Heartbeat maybe fails to properly keep connection punched (this is discussed on reddit rustdesk but not here I believe)

But once the service is enabled, it works well with unattended.

**Comment by rustdesk** [comment]
Please try out it tomorrow, just added.

**Comment by michalwitek1232** [comment]
@rustdesk  thanks a lot, you and your team are great :+1:

**Comment by homeusepb** [comment]
> Please try out it tomorrow, just added.

Beautiful! It looks good, I am just trying it out. Seems to do the job.
1 more thing we can do to make it more stable (quick fix). It auto starts on boot but does not auto start on application launch/if application was force closed/killed, it does not auto start. This should be an easy fix and whenever the application is ran (foreground or background), just launch it the same way you launched it on boot. This should fix it fully.

**Comment by rustdesk** [comment]
> but does not auto start on application launch/if application was force closed/killed, it does not auto start.

@Heap-Hop check if anydesk has this behavior please.

**Comment by homeusepb** [comment]
Yeah basically, currently RustDesk listens to "ON_BOOT_RECEIVE_COMPLETED" intent by Android so it works on boot.
We can also add one more logic of code with ON App launched (easier because Rustdesk can do this internally)
:)

**Comment by rustdesk** [comment]
close as fix merged

**Comment by peterscafe** [comment]
Hi There, i was testing the nightly build fot th start on boot feature on and Android 9 TVBox, but the 2 enhancements settings keep rustdesk background service and start on boot are disable and unable to activate. Is it Because of the Android 9 version ?

tks

**Comment by yoad-m-pp** [comment]
@rustdesk How to set the Auto start on boot settings with a custom apk if the disables settings is toggled on?

**Comment by LordSithek** [comment]
So, there is still no any other intent to start server except BOOT_COMPLETED? I know I can use it eg. in Tasker but it requires Autostart to be enabled and I don't really want it to start at boot...
